contracts/peripheral/PriceConverterOracleWrapper.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.10;
+
+import {AggregatorV3Interface} from "../interfaces/IAggregatorV3.sol";
+
+
+/**
+    @title Oracle wrapper
+    @notice Oracle wrapper that uses two oracles to return price in a differen denomination
+ */
+contract PriceConverterOracleWrapper is AggregatorV3Interface {
+
+    AggregatorV3Interface public immutable basePriceOracle;
+    AggregatorV3Interface public immutable quotePriceOracle;
+    uint8 public immutable override decimals;
+    string public constant override description = "Oracle wrapper that uses two oracles to return price in a differen denomination";
+    uint256 public constant override version = 0;
+
+
+    constructor(address _basePriceOracleAddress, address _quotePriceOracleAddress, uint8 decimals_) {
+        basePriceOracle = AggregatorV3Interface(_basePriceOracleAddress);
+        quotePriceOracle = AggregatorV3Interface(_quotePriceOracleAddress);
+        decimals = decimals_;
+    }
+
+    function latestRoundData() external view returns (
+      uint80 roundId,
+      int256 answer,
+      uint256 startedAt,
+      uint256 updatedAt,
+      uint80 answeredInRound
+    ) {
+        ( , int256 basePrice, , , ) = AggregatorV3Interface(basePriceOracle).latestRoundData();
+        uint8 baseDecimals = AggregatorV3Interface(basePriceOracle).decimals();
+        basePrice = scalePrice(basePrice, baseDecimals, decimals);
+
+        ( , int256 quotePrice, , , ) = AggregatorV3Interface(quotePriceOracle).latestRoundData();
+        uint8 quoteDecimals = AggregatorV3Interface(quotePriceOracle).decimals();
+        quotePrice = scalePrice(quotePrice, quoteDecimals, decimals);
```

Comment:
i know what you mean because i also noticed a rounding error in the way this code does things. i coped this from 
https://docs.chain.link/docs/get-the-latest-price/#getting-a-different-price-denomination

which yacademy linked me

Reply:
the rounding error should be not that significant but it is nice fixing it i'l see what fixed version looks like and see if worth redoing it

Reply:
ok i had a look more closely. rounding error will only happen if we want to go from higher decimals to lower. Let say we have 2 oracles at 18 decimals but we want the aggregated price to be in 9 decimals. In this case it would cause rounding error.

Reply:
ended up reimplementing it anyway since the reimplementation is quite clean and easier to read imo